purchases/src/androidTestIntegrationTest/kotlin/com/revenuecat/purchases/BasePurchasesIntegrationTest.kt

```diff
@@ -0,0 +1,118 @@
+package com.revenuecat.purchases
+
+import android.content.Context
+import android.preference.PreferenceManager
+import androidx.test.ext.junit.rules.activityScenarioRule
+import com.revenuecat.purchases.common.BillingAbstract
+import com.revenuecat.purchases.models.StoreTransaction
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.slot
+import org.assertj.core.api.Assertions
+import org.junit.Rule
+import java.net.URL
+import java.util.Date
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+import kotlin.time.Duration.Companion.seconds
+
+open class BasePurchasesIntegrationTest {
+
+    @get:Rule
+    var activityScenarioRule = activityScenarioRule<MainActivity>()
+
+    protected val testTimeout = 5.seconds
+    protected val currentTimestamp = Date().time
+    protected val testUserId = "android-integration-test-$currentTimestamp"
+    protected val proxyUrl = Constants.proxyUrl.takeIf { it != "NO_PROXY_URL" }
+
+    protected lateinit var mockBillingAbstract: BillingAbstract
+
+    protected var latestPurchasesUpdatedListener: BillingAbstract.PurchasesUpdatedListener? = null
+    protected var latestStateListener: BillingAbstract.StateListener? = null
+
+    // region helpers
+
+    protected fun setupTest(
+        initialSharedPreferences: Map<String, String> = emptyMap(),
+        initialActivePurchases: Map<String, StoreTransaction> = emptyMap(),
+        postSetupTestCallback: (MainActivity) -> Unit = {}
```

Comment:
This was added because `onActivityReady` wasn't guaranteed to be executed in order which resulted in some flaky tests. This way, we execute the test setup code and the test itself within a single `onActivityReady`.